src/Rest/Routes/Integrations/Paycek/FormSubmitPaycekRoute.php

```diff
@@ -0,0 +1,244 @@
+	{
+		return '/' . UtilsConfig::ROUTE_PREFIX_FORM_SUBMIT . '/' . self::ROUTE_SLUG;
+	}
+
+
+	/**
+	 * Implement submit action.
+	 *
+	 * @param array<string, mixed> $formDetails Data passed from the `getFormDetailsApi` function.
+	 *
+	 * @return mixed
+	 */
+	protected function submitAction(array $formDetails)
+	{
+		$formId = $formDetails[UtilsConfig::FD_FORM_ID];
+
+		if (!\apply_filters(SettingsPaycek::FILTER_SETTINGS_IS_VALID_NAME, $formId)) {
+			return \rest_ensure_response(
+				UtilsApiHelper::getApiErrorPublicOutput(
+					$this->labels->getLabel('paycekMissingConfig', $formId)
+				)
+			);
+		}
+
+		$mapParams = UtilsSettingsHelper::getSettingValueGroup(SettingsPaycek::SETTINGS_PAYCEK_PARAMS_MAP_KEY, $formId);
+
+		$params = $this->prepareParams($mapParams, $formDetails['paramsRaw'], $formId);
+
+		$reqParams = [
+			'profileCode',
+			'secretKey',
+			'paymentId',
+			'amount',
+		];
+
+		$missingOrEmpty = \array_filter($reqParams, fn($param) => empty($params[$param] ?? null));
+
+		if ($missingOrEmpty) {
+			return \rest_ensure_response(
+				UtilsApiHelper::getApiErrorPublicOutput(
+					$this->labels->getLabel('paycekMissingReqParams', $formId)
+				)
+			);
+		}
+
+		// Finish.
+		return \rest_ensure_response(
+			UtilsApiHelper::getApiSuccessPublicOutput(
+				$this->labels->getLabel('paycekSuccess', $formId),
+				[
+					UtilsHelper::getStateResponseOutputKey('processExternally') => [
+						'type' => 'GET',
+						'url' => $this->generatePaymentUrl(
+							$params['profileCode'],
+							$params['secretKey'],
+							$params['paymentId'],
+							$params['amount'],
+							$params['email'],
+							$params['description'],
+							$params['language'],
+							$params['urlSuccess'],
+							$params['urlFail'],
+							$params['urlCancel']
+						),
+					],
+				]
+			)
+		);
+	}
+
+	/**
+	 * Prepare params for Paycek.
+	 *
+	 * @param array<string, string> $mapParams Map of params.
+	 * @param array<string, string> $params Form params.
+	 * @param string $formId Form ID.
+	 *
+	 * @return array<string, string>
+	 */
+	private function prepareParams(array $mapParams, array $params, string $formId): array
+	{
+		$output = [];
+
+		foreach ($mapParams as $key => $value) {
+			$param = $params[$value] ?? '';
+			if (!$param) {
+				continue;
+			}
+
+			switch ($key) {
+				case 'amount':
+					$param = \number_format((float)$param, 2, '.', '');
+					break;
+			}
+
+			$output[$key] = $param;
+		}
+
+		$output['secretKey'] = UtilsSettingsHelper::getSettingsDisabledOutputWithDebugFilter(Variables::getApiKeyPaycek(), SettingsPaycek::SETTINGS_PAYCEK_API_KEY_KEY)['value'];
+		$output['profileCode'] = UtilsSettingsHelper::getSettingsDisabledOutputWithDebugFilter(Variables::getApiProfileKeyPaycek(), SettingsPaycek::SETTINGS_PAYCEK_API_PROFILE_KEY)['value'];
+		$output['language'] = UtilsSettingsHelper::getSettingValue(SettingsPaycek::SETTINGS_PAYCEK_LANG_KEY, $formId);
+		$output['paymentId'] = 'order_' . \time();
```

Comment:
Could this be a problem if in theory 2 orders come in at the exact same time?

Reply:
in theory yes